server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexStateServiceBatchingTests.java

```diff
@@ -118,7 +119,7 @@ public void testBatchCloseIndices() throws Exception {
         block2.run(); // wait for block
 
         // wait for the queue to have the second close tasks (the close-indices tasks)
-        assertBusy(() -> assertThat(masterService.pendingTasks(), hasSize(3))); // two plus the blocking task itself
+        assertBusy(() -> assertThat(findPendingTasks(masterService, "close-indices"), hasSize(2)));
```

Comment:
Under heavy CPU load "reconcile-desired-task" could have been delayed and sometimes this busy block would wait for the first 3 tasks that would be "reconcile-desired-task", "block" and only one of the close tasks. Since the second task is missed the assertion for batching two task is not satisfied.